### PR TITLE
[MENFORCER-469] Fix banTransitiveDependencies and transitive dependen…

### PR DESCRIPTION
**Pull request by slawekjaranowski**
…cies with another version than the resolved one

---
- introduce methods:

  - resolveTransitiveDependenciesVerbose resolve with full tree contains all dependencies - also conflicted

  - resolveTransitiveDependencies resolve final tree contains dependencies after conflict resolved

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java [hunk 1]
```diff
@@ -66,6 +62,24 @@ class ResolveUtil {
...
 
+    /**
+     * Retrieves the {@link DependencyNode} instance containing the result of the transitive dependency
+     * for the current {@link MavenProject} in verbose mode.
+     * <p>
+     * In verbose mode all nodes participating in a conflict are retained.
+     * </p>
+     * <p>
+     * Please consult {@link ConflictResolver} and {@link DependencyManagerUtils}>
+     * /p>
```
Comment (elharo) [inline enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java:73]: `</p>`

File: enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java [hunk 2]
```diff
@@ -66,6 +62,24 @@ class ResolveUtil {
         this.session = Objects.requireNonNull(session);
     }
 
+    /**
+     * Retrieves the {@link DependencyNode} instance containing the result of the transitive dependency
```
Comment (elharo) [inline enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java:66]: I'm not sure what "the result of the transitive dependency" is exactly. Is this just the transitive dependencies of the current maven project? Consider rephrasing.

Comment (slawekjaranowski) [reply enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java:66]: Yes it is for current maven project

Comment (elharo) [reply enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java:66]: Please rewrite to clarify this

Comment (slawekjaranowski) [reply enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java:66]: please look at next lines - there is:

```
Retrieves the {@link DependencyNode} instance containing the result of the transitive dependency
for the current {@link MavenProject} in verbose mode.
```

Comment (elharo) [reply enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java:66]: "the result of the transitive dependency" is unclear. Transitive dependencies are not  process and wouldn't normally be thought of as having results.

Comment (slawekjaranowski) [reply enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/dependency/ResolveUtil.java:66]: So ... can you propose description of this method?

**Comment by jira-importer** [comment]
Resolve #842

**Comment by jira-importer** [comment]
Resolve #842

**Comment by jira-importer** [comment]
Resolve #842

**Pull request merged**